### PR TITLE
TST: Prototype of notebook viz (panel)

### PR DESCRIPTION
**Pull request by GuillaumeFavelier**
This PR implements a prototype of integration of the pyvista backend in the jupyter notebook.

Animation:
output

Static images:
image
image
image
image
image

*Note: requires the `panel` package*

### CODE REVIEW

**Comment by larsoner** [comment]
Awesome!

Does it work with a multi-view scene, e.g. `stc.plot(hemi='split', views=['lat', 'med'])`? If not, is it possible in the future or would it be a nightmare or technically impossible to implement?

**Comment by GuillaumeFavelier** [comment]
Unfortunately, it's disabled for now. The output will just be the output image. Multi-view will be challenging to add I think so I can't say for now but I don't exclude it since `viewport` management is normally supported in [vtk.js](https://kitware.github.io/vtk-js/api/Rendering_Core_Viewport.html).

File: mne/viz/backends/_pyvista.py [hunk 1]
```diff
@@ -395,7 +399,8 @@ def scalarbar(self, source, title=None, n_labels=4, bgcolor=None):
                                         background_color=bgcolor)
 
     def show(self):
-        self.figure.display = self.plotter.show()
+        self.figure.display = self.plotter.show(use_panel=True,
+                                                auto_close=False)
```
Comment (agramfort) [inline mne/viz/backends/_pyvista.py]: will this break outside of notebook?

Comment (GuillaumeFavelier) [reply mne/viz/backends/_pyvista.py]: Oh my! You're right. I will fix this

**Comment by GuillaumeFavelier** [comment]
`panel` 0.7 is out with some [enhancements](https://github.com/pyvista/pyvista/pull/268#issuecomment-556034711). I'll update the renderings accordingly after testing more.

**Comment by GuillaumeFavelier** [comment]
This [panel example](https://panel.pyviz.org/reference/panes/VTK.html#gallery-vtk) demonstrates multi-view with linked views @larsoner

**Comment by larsoner** [comment]
Awesome!

**Comment by akaszynski** [comment]
@GuillaumeFavelier, this is perfect.  Nice work!  I'd like to have multiple plotting backends supported for `pyvista` as they each carry their own advantages, and some don't work on certain platforms (i.e. `itk` doesn't work on iOS).

Perhaps adding a `backend` kwarg to the `plot` and `show` methods?  That way we can open it up to multiple plotting backends and leave `panel` as default.

**Comment by GuillaumeFavelier** [comment]
Although I agree no backend is perfect, the cost of maintaining this multi-backend architecture plus ensuring the same standard of quality for all of them can be huge in my opinion. I would recommend to stick with the backend with the best coverage of the needed features and then work hard to fill the gap.

**Comment by jona-sassenhagen** [comment]
The GIF looks great.

**Comment by GuillaumeFavelier** [comment]
Closing now since we got #7758 working

**Pull request closed without merging**